asciidoc/courses/drivers-java/modules/1-driver/lessons/4-object-mapping/lesson.adoc

```diff
@@ -0,0 +1,139 @@
+= Mapping Results to Java Objects
+:type: lesson 
+:slides: true
+:minutes: 15
+:order: 5
+
+[.slide.col-2]
+== Introduction
+
+[.col]
+As of the Neo4j Java driver 5.28.5, the driver includes an link:https://github.com/neo4j/neo4j-java-driver/discussions/1640[object mapping feature] that allows you to map query results directly to Java objects. This feature simplifies the process of working with Neo4j data in Java applications by eliminating the need for manual mapping with raw data types.
+
+[.col]
+====
+[NOTE]
+.Object Graph Mapping (OGM)
+=====
+The driver's new object mapping is not a full-fledged OGM solution. For a more comprehensive option, check out the https://neo4j.com/docs/ogm-manual/current/[Neo4j OGM library^].
+=====
+====
+
+[.slide]
+== Domain Model
+
+For a person domain class, create a new file in `src/main/java/com/neo4j/app` and name it `Person.java`. This example will use Java records (rather than class), but classes works similarly.
+
+[source,Java]
+----
+public record Person(String id, 
+                     String name) {
+}
+----
+
+[.slide.col-2]
+== Querying and Mapping Results
+
+[.col]
+====
+Next, open the `src/main/java/com/neo4j/app/App.java` file and add a method to query for a person by name and return an entity mapped to your `Person` record:
+
+[source,Java]
+----
+final String personCypher = """
+    MATCH (person:Person {name: $name})
+    RETURN person
+    """;
+final String name = "Tom Hanks";
+
+var person = driver.executableQuery(personCypher)
+    .withParameters(Map.of("name", name))
+    .execute()
+    .records()
+    .stream()
+    .map(record -> record.get("person").as(Person.class)) // <1>
+    .findFirst()
+    .orElseThrow(() -> new RuntimeException("Person not found")); // <2>
+
+System.out.println(person); // <3>
+----
+====
+
+[.col]
+====
+<1> `.as(Person.class)` returns a single record mapped to the `Person` class
+<2> `.findFirst().orElseThrow()` returns the first record or throws an exception if not found
+<3> Print the person object to the console
+====
+
+[.slide]
+== Adding a Connecting Node
+
+Now add the connecting `Movie` node to the domain. This will connect the `Movie` node to the `Person` node, returning a graph.
```

Comment:
```suggestion
You can return a graph by adding the `Movie` node to the domain and connecting it to the `Person` node.
```

Does this still work? I wanted to move to active voice. I tend to avoid the use of Now, let's, etc.